[Ravi Ananya] iP

Banana3021 opened the pull request:
# Banana :banana:

## Welcome to Banana, the new and improved Duke chatbot!

> Why do you need to write down tasks when you've got a chatbot.

### Here's what Banana can do:
1. Store your task information
1. Retrieve your task information immediately
1. Access your task information
    1. Find a task
    1. Delete a task
    1. Edit a task to indicate that it has been completed
    1. Use a proper `<Tasklist>` element

### It's super cool! 
* User-friendly
* Really easy to run
* Checks off tasks in a satisfying way.

### The next list of features:
- [x] GUI
- [ ] More types of tasks.

### Download the code from [GitHub](https://github.com/Banana3021/ip)

### How to run the code
```javascript
     public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
```

Banana3021 committed:
level 1 modified

Banana3021 committed:
level 1, 2 modified

Banana3021 committed:
level 1, 2 modified

Banana3021 committed:
level 1-3 modified

Banana3021 committed:
Level 1, 2added

Banana3021 committed:
level 1 added

Banana3021 committed:
level 1-4 finished

Banana3021 committed:
level 5, 6 added

Banana3021 committed:
code tidied up

Banana3021 committed:
Levels 1-6 completed

Banana3021 committed:
Add level 7 code

Banana3021 committed:
recent commit

Banana3021 committed:
Merge master into Level 7 branch

Banana3021 committed:
commit from master after merge

Banana3021 committed:
Merge branch 'branch-Level-7'

Banana3021 committed:
Add level 7 changes

Banana3021 committed:
Add placement changes.

Banana3021 committed:
Add Level 8

Banana3021 committed:
Add new constant names.

Banana3021 committed:
Merge branch 'master' into branch-Level-8

Banana3021 committed:
Add indents.

Banana3021 committed:
Revert "Add indents."

This reverts commit df4626b0d00a7f148e1fd5220f744a1b8fa9a113.

Banana3021 committed:
Add indents.

Banana3021 committed:
Add indents and edited code.

Banana3021 committed:
Merge branch 'branch-Level-8'

Banana3021 committed:
Keep necessary code.

Banana3021 committed:
Add Parser, Storage, Ui and Tasklist classes

Banana3021 committed:
Add Parser, Storage, TaskList and Ui classes

Banana3021 committed:
Revert "Add Parser, Storage, TaskList and Ui classes"

This reverts commit 2da7d4127980d769c078f5b5f3341b577d641a2b.

Banana3021 committed:
Merge branch 'master' of https://github.com/Banana3021/ip

Banana3021 committed:
Add changes to Duke, Parser, Storage, TaskList and Ui classes

Banana3021 committed:
Add package statement

Banana3021 committed:
Add test class

Banana3021 committed:
Add indentation.

Banana3021 committed:
Add indents.

Banana3021 committed:
Add more format edits.

Banana3021 committed:
Add format edits.

Banana3021 committed:
Add format changes.

Banana3021 committed:
Add more format changes.

Banana3021 committed:
Merge branch 'master' into branch-Level-9

Banana3021 committed:
Add new formatting.

Banana3021 committed:
Add even more formats.

Banana3021 committed:
Add some more formatting.

Banana3021 committed:
Merge branch 'branch-Level-9'

Banana3021 committed:
Add more formats.

Banana3021 committed:
Add some indents.

Banana3021 committed:
Add some more indents.

Banana3021 committed:
Merge branch 'master' into A-JavaDoc

Banana3021 committed:
Add new comments.

Banana3021 committed:
Add some more new comments.

Banana3021 committed:
Add more comments.

Banana3021 committed:
Merge branch 'A-JavaDoc'

Banana3021 committed:
Add more comment info.

Banana3021 committed:
Add more info.

Banana3021 committed:
Add more commenting.

Banana3021 committed:
Add more comments.

Banana3021 committed:
Merge branch 'master' into A-CodingStandard

Banana3021 committed:
Add even more comments info.

Banana3021 committed:
Add even more info.

Banana3021 committed:
Merge branch 'A-CodingStandard'

Banana3021 committed:
Add more info.

Banana3021 committed:
Add jar file.

Banana3021 committed:
Add more class comments.

Banana3021 committed:
Add code format edits.

Banana3021 committed:
Add new comments.

Banana3021 committed:
Add new test cases.

Banana3021 committed:
Add gradle build.

Banana3021 committed:
Add gradle files.

Banana3021 committed:
Merge branch 'master' into A-Gradle

Banana3021 committed:
Add code edits.

Banana3021 committed:
Add file changes.

Banana3021 committed:
Merge branch 'A-Gradle'

Banana3021 committed:
Add GUI.

JunWei3112 commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,352 @@
+public class Duke {
+
+    private static final String WELCOME_LABEL = "Hello! I'm Banana \n" + "     What can I do for you?";
+    private static final String BYE_LABEL = "Bye. Hope to see you again soon!";
```
Should these associated constants be prefixed by the common type name instead?
```suggestion
    private static final String LABEL_WELCOME = "Hello! I'm Banana \n" + "     What can I do for you?";
    private static final String LABEL_BYE = "Bye. Hope to see you again soon!";
```

JunWei3112 commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,352 @@
+    private static final String WELCOME_LABEL = "Hello! I'm Banana \n" + "     What can I do for you?";
+    private static final String BYE_LABEL = "Bye. Hope to see you again soon!";
+
+    private Storage store;
```
Perhaps it will be better if a noun was used for this variable name, instead of using a verb (which is better suited for methods/functions) ? 
```suggestion
    private Storage storage;
```

JunWei3112 commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,352 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);*/
```
Could this chunk of commented out code (also known as **dead code**) be deleted, since they are redundant?

JunWei3112 commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,352 @@
+ **/
+class DukeException extends Exception {
+
+    private String text;
```
Perhaps it would be better if a more descriptive name (e.g. errorMessage) was used for this variable?

JunWei3112 commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,352 @@
+
+class Deadline extends Task {
+
+    protected String deadLine;
```
Should this variable be renamed to **deadline** instead, since deadline is a single word and naming it deadLine may cause confusion?

JunWei3112 commented on src/main/java/banana/Parser.java:
```diff
@@ -0,0 +1,281 @@
+     * @param tasks the list of tasks.
+     * @throws DukeException if the input was invalid.
+     */
+    public void useInput(TaskList tasks) throws DukeException {
```
Since there are many ways in which input can be used, should a more descriptive name be used for this method so as to further specify the purpose of this method? One possible suggestion is **parseInput**

Banana3021 committed:
add more info.

Banana3021 committed:
Add comment changes.

JunWei3112 commented on src/main/java/banana/Parser.java:
```diff
@@ -0,0 +1,281 @@
+     * @return the date.
+     */
+    public String parseDates(String date) {
+        String[] sep = date.split("/");
```
As readers may not be able to infer the purpose of this variable from the abbreviated variable name, perhaps it would be better if a lengthier name was used to describe this variable instead? I noticed this same issue in some other places as well.

Banana3021 committed:
Add commit reordering.

JunWei3112 commented on src/main/java/banana/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+     *
+     * @return the size of the list.
+     */
+    public int size() {
```
Should verbs be used in the naming of this method instead? One possible suggestion is 
```suggestion
    public int getSize() {
```

Banana3021 committed:
Remove commented out code.

JunWei3112 commented on src/main/java/banana/Ui.java:
```diff
@@ -0,0 +1,44 @@
+
+public class Ui {
+
+    protected Scanner userInput;
```
Should another name be used in naming this Scanner variable instead? Naming it **userInput** may cause confusion to readers, since the purpose of this variable is to read and obtain user input, and readers may misinterpret that this variable is used to store user input instead.

Banana3021 committed:
Merge branch 'master' into branch-Level-10

Banana3021 committed:
Add changes.

Banana3021 committed:
Merge branch 'branch-Level-10'

JunWei3112 reviewed (commented):
Overall, I found that your code was quite easy to read and your methods also contain an appropriate number of lines of code (not too lengthy and nested). 
However, some of the classes may be getting too long. One example will be Duke.java, where classes that serve other purposes are also being declared in the same file. You can consider splitting it into different class files.
Also, do take extra note when naming variables or methods, so as to improve readability of the code. Otherwise, good job! 😁

Banana3021 committed:
Deleted some folders.

Banana3021 committed:
Add more comments.

Banana3021 committed:
Add more comments

Banana3021 committed:
Add structure changes

Banana3021 committed:
Add assert statements

Banana3021 committed:
Modify assert statements

Banana3021 committed:
Merge pull request #1 from Banana3021/A-Assertions

Add and modify assertions

rgbpokka commented on src/main/java/banana/Duke.java:
```diff
@@ -0,0 +1,251 @@
+    }
+
+}
+
```
Overall I feel the code is well organized into distinct methods, however it is a very large file. Perhaps the writeToFile method could be part of the Storage class instead?

rgbpokka commented on src/main/java/banana/Parser.java:
```diff
@@ -0,0 +1,294 @@
+        } else {
+            tasks.addTask(new Task(input));
+        }
+        return displayLabel("Got it. I've added this task:  \n" +
```
If i'm not wrong, the line should break before an operator (the "+" sign should go onto the start of the next line instead of being at the end of the line)

rgbpokka commented on src/main/java/banana/Parser.java:
```diff
@@ -0,0 +1,294 @@
+     * @return the date.
+     */
+    public String parseDates(String date) {
+        String[] sepIntoYearMonthDate = date.split("/");
```
I like the clear variable name.

rgbpokka reviewed (commented):
Overall I think the code is really clean and easy to read! Keep up the good work :)

Banana3021 committed:
Improve code quality

Banana3021 committed:
Merge pull request #2 from Banana3021/branch-A-CodeQuality

Improve code quality

Banana3021 committed:
Add more changes

Banana3021 committed:
Add extension

Banana3021 committed:
Parser class: parses the user’s input and provides the according output

A new command called "undo" helps to reverse the input that was just
made, but its edge cases need to be tweaked.

Let’s check for the previous command is null while simultaneously
checking whether the TaskList is empty.

This is to prevent an exception from being thrown if
"undo" is the user's first input, as a result of the previous input
not yet holding a value.

Banana3021 committed:
Revert "Parser class: parses the user’s input and provides the according output"

This reverts commit c027e8fa2cfdcc77085bec24a72d2bb261396484.

Banana3021 committed:
Parser class: parses the user’s input and provides the according output

A new command called "undo" helps to reverse the input that was just
made, but its edge cases need to be tweaked.

Let’s check whether the previous command is null.

This is to prevent an exception from being thrown if
"undo" is the user's first input, as a result of the previous input
not yet holding a value.

Banana3021 committed:
Parser classes: same function, to decipher the user's input.

The Parser class, with several functions to handle all the commands,
has become too long.

Let’s create another class called Parser functions, to remove out any
functions that do not directly deal with the user input at first. These
functions do not have Command as part of their name. For example,
getItems is only dealt with in Parser’s listCommand function, and can
therefore be relegated to a separate class.

This helps to reduce the size of the Parser class.

Banana3021 committed:
Add ParserFunctions class

Banana3021 committed:
Add FXML feature

Banana3021 committed:
Add user guide

Banana3021 committed:
Add representative screenshot

Banana3021 committed:
Edit Task list

Banana3021 committed:
Add title to the stage

Banana3021 committed:
Edit welcome label and add scrollPane features

Banana3021 committed:
Edit layout of the program and change welcome label

Banana3021 committed:
Edit README.md

Banana3021 committed:
Improve GUI features by changing colors and fonts

Banana3021 committed:
Add comments to the DialogBox class

Banana3021 committed:
Edit comments in Main class

Banana3021 committed:
Edit comments and remove print statement

Banana3021 committed:
Move images to a different folder

Banana3021 committed:
Edit Javadoc comments and format code

Banana3021 committed:
Separate the Duke and DukeException classes

Banana3021 committed:
Edit comments and code

Banana3021 committed:
Edit Tasks.txt file

Banana3021 committed:
Change product photo

Banana3021 committed:
Change text size of user input

Banana3021 committed:
Remove images from old location

Banana3021 committed:
Change Tasks.txt folder when running the program

Banana3021 committed:
Modify README.md

Banana3021 committed:
Remove assert statements and add exception case

Banana3021 committed:
Change welcome label

Banana3021 committed:
Modify Javadoc comments

Banana3021 committed:
Modify Javadoc comments and change label font size

Banana3021 committed:
Change user input handling function

Banana3021 committed:
Remove main method

Banana3021 committed:
Change return description for parseDates method

Banana3021 committed:
Remove unnecessary code and import statements

Banana3021 committed:
Change access modifiers

Banana3021 committed:
Rearrange instance variable ordering

Banana3021 committed:
Retake program image

Banana3021 committed:
Correct exception condition

Banana3021 committed:
Add function to check for invalid index

Banana3021 committed:
Modify how to handle exceptions

Banana3021 committed:
Change exception message

Banana3021 committed:
Edit the findCommand method

Banana3021 committed:
Edit Tasks.txt file

Banana3021 committed:
Edit path of image files

Banana3021 committed:
Use relative instead of absolute file path

Banana3021 committed:
Move Tasks.txt file

Banana3021 committed:
Change exception to be caught

Banana3021 committed:
Write method to create file and directory

Banana3021 committed:
Change exception condition
